src/Compilers/CSharp/Portable/Lowering/Instrumentation/DynamicAnalysisInjector.cs

```diff
@@ -129,6 +129,31 @@ private static bool IsExcludedFromCodeCoverage(MethodSymbol method)
             return false;
         }
 
+        private BoundExpression GetFileIndices()
+        {
+            var documents = PooledHashSet<DebugSourceDocument>.GetInstance();
+            foreach (var span in _dynamicAnalysisSpans)
+            {
+                documents.Add(span.Document);
+            }
+
+            if (documents.Count == 0)
```

Comment:
>if (documents.Count == 0) [](start = 12, length = 25)

This can happen only if _dynamicAnalysisSpans is empty, therefore, the check can be performed before we even try to allocate the HashSet.